### DISCUSSION TITLE
Version 1.0 ?

### DISCUSSION THREAD

**Discussion by jhillairet** [scikit-rf/scikit-rf#692] [Ideas]
Some potential users said to me that looking at `scikit-rf`, they saw the version number as `0.XY` and they deduced that the code was highly unstable, then drop and didn't look it further.

So I wonder if at some point we should jump to a `1.0` version and revise the way the version numbers are increased. Any thoughts or ideas about that? 

Update: See issue #718

**Comment by arsenovic** [comment]
wow. i didnt know people were so prejudice against <1.0. if this is a problem i would definitely  think changing to 1.0 is a good idea.

**Comment by FranzForstmayr** [comment]
A Version 1.0 jump could help us to remove some parts of the code which aren't used and hard to maintain. After then a semantic versioning scheme could be used to help users managing deprecated parts of the api.

**Comment by FranzForstmayr** [comment]
I'll start a controversial topic right now...

skrf has lot's of APIs which are hard coded to GHz or nanoseconds. 
The default frequency scaling in GHz created lots of confusion for me and other users.
Would version 1.0 be a chance to get rid of this predefined scaling and use hertz and seconds consistently?

As a migration path we should warn the users which are still using GHz by default to explicitly use the "unit" or "f_unit" argument for a few versions of scikit-rf. Version 1.0 will interprete a undefined frequency as Hz.

**Reply by Vinc0110** [reply]
I'm all in favor of changing the defaults to proper SI units. I find it error-prone to specify in nanoseconds instead of ...e-9 seconds.

**Reply by jhillairet** [reply]
I also agree that it would be more logical to adopt SI units... It's just we will need a couple of version to get users properly warned about a potential major change in their code results...

**Reply by mhuser** [reply]
SI units defaults sounds like music to me !
